README.md

```diff
@@ -1,125 +1,130 @@
+~~~terminal
+    git clone git@github.com:Travelinglight/iknn-HA.git
+    cd iknn-HA
+~~~
+
+### 2. Import LPinit.sql (in postgresql)
+
+~~~sql
+    \i pgsql/HAinit.sql
+~~~
+
+### 3. Initialize a target table to support iknn query
+You may want to use the sample table "test". Import the table before you initialize it:
+
+~~~sql
+    CREATE DATABASE iknn;
+    \i iknn.sql
+~~~
+
+Now initialize the "test" table:
+
+~~~sql
+    select hainit('test', 2,4,3,4);
+~~~
+
+Command format:
+
+* 'test' is the table name
+* 2,4,3,4 are the number of bins on each field. The table 'test' has four columns, so there are four numbers following the table name.
+
+The hainit function automatically does these things:
+
+  1. Create a tmp table with column dimension, nbin, nobj;
+  2. Add a column to the original table: ha_id, recording the unique id for ha algorithm;
+  3. Create table for each bin, representing buckets, with the name habin\_[table name]\_[field name]\_id. e.g. habin\_test\_a0\_1. Each bin has two columns: val and ha_id;
+  4. Build up b-tree index on habin\_[table name]\_[field name]\_id at column val, to auto-sort the tuples with field value;
+  5. Distribute objects into bins;
+  7. Set triggers to maintain the three columns and the extra tables on insert, update and delete.
+
+### 4. Make and install iknnLP function (in terminal)
+~~~terminal
+    cd c
+    make
+    sudo make install
+    cd ..
+~~~
+
+### 5. Import iknnLP function (in postgresql)
+~~~sql
+    \i c/iknnHA.sql
+~~~
+
+### 6. Performing iknn query with LP althrothm
+~~~sql
+    select a, b, c, d, distance from iknnHA('find 3 nearest neighbour of (a0,a2,a3)(31,33,34) from test') AS (a int, b int, c int, d int, distance float);
+~~~
+* a0,a2,a3 are columns in the table _test_.
+* 31,33,34 are values of the columns respectively.
+* The tuples returned are those considered nearest with the query object.
+
+### 7. Here's the result
+~~~sql
+ a | b  | c  | d  | distance 
+---+----+----+----+----------
+   | 82 | 43 | 38 |      232
+   | 17 |    | 35 |        4
+   |  2 |    | 39 |      100
+(3 rows)
+~~~
+
+### 8. Inport LPwithdraw.sql
+
+~~~
+    \i HAwithdraw.sql
+~~~
+
+### 9. Withdraw iknn query support
+
+~~~
+    select hawithdraw([table name]);
+~~~
+
+This function automatically does these things:
+  1. Drop the extra column (ha_id) of the original table;
+  2. Drop all tmp tables;
+  3. Drop the three triggers that maintain the tmp table and the bins;
+
+## Q&A
+### I cannot create hstore extension when importing LPinit.sql.
+  In ubuntu, you need to install the contrib before you create them:
+
+~~~
+    sudo apt-get install postgresql-contrib-9.4
+~~~
+
+  Or you can install postgresql with those contribs:
+
+~~~
+    sudo apt-get install postgresql postgresql-contrib
+~~~
+
+## Contact us
+1. You can get the paper from Mr. Gao: gaoyj@zju.edu.cn
+2. The projet is coded by Kingston Chen. Feel free to ask any question: holaelmundokingston@gmail.com

```

Comment:
Just googled this change: 

> When any of is followed by a countable plural noun, the verb can be in either singular or plural form, but a singular verb is more common in a formal style

Just in case you don't know, and thanks for this catch.